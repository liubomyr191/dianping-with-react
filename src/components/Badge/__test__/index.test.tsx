import React from 'react'
import { View, Text } from 'react-native'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { Badge }from '../'

configure({ adapter: new Adapter() })

describe('Badge Snapshot', () => {
  test('renders default type', () => {
    const props = { title: '测试tag' }
    const wrapper = shallow(
      <Badge { ...props } />
    )
    expect(wrapper).toMatchSnapshot()
  })
})

describe('Badge', () => {
  test('it render default type with "text"', () => {
    const wrapper = shallow(
      <Badge/>
    )
    expect(wrapper.instance().props).toEqual({ type: 'text' })
  })

  test('it render dot correctly', () => {
    const wrapper = shallow(
      <Badge type={'dot'}/>
    )
    expect(wrapper.instance().props).toEqual({ type: 'dot' })
  })

  test('it render triangle correctly', () => {
    const wrapper = shallow(
      <Badge type={'triangle'} title={'优惠'} />
    )
    expect(wrapper.instance().props).toEqual({ type: 'triangle', title: '优惠' })
  })

  test('it can set title with number type', () => {
    const wrapper = shallow(
      <Badge title={111} />
    )
    expect(wrapper.instance().props).toEqual({ title: 111, type: 'text' })
  })

  test('it can render slot correctly', () => {
    const wrapper = shallow(
      <Badge title={111} >
        <View><Text>aaa</Text></View>
      </Badge>
    )
    expect(wrapper.contains(<View><Text>aaa</Text></View>)).toEqual(true)
    expect(wrapper.contains(<View><Text>111</Text></View>)).toEqual(false)
  })

})
