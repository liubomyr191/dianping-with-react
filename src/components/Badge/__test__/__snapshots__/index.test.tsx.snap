// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Snapshot renders default type 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Badge
    title="测试tag"
    type="text"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#FF301A",
                "borderRadius": 12,
                "justifyContent": "center",
                "paddingHorizontal": 6,
                "paddingVertical": 2,
              },
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontSize": 10,
                },
                undefined,
              ]
            }
          >
            测试tag
          </Text>
        </View>,
        null,
        null,
      ],
      "style": Object {
        "flexDirection": "row",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "fontSize": 10,
                  },
                  undefined,
                ]
              }
            >
              测试tag
            </Text>,
            undefined,
          ],
          "style": Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#FF301A",
              "borderRadius": 12,
              "justifyContent": "center",
              "paddingHorizontal": 6,
              "paddingVertical": 2,
            },
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "测试tag",
              "ellipsizeMode": "tail",
              "style": Array [
                Object {
                  "color": "#fff",
                  "fontSize": 10,
                },
                undefined,
              ],
            },
            "ref": null,
            "rendered": "测试tag",
            "type": [Function],
          },
          undefined,
        ],
        "type": [Function],
      },
      null,
      null,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FF301A",
                  "borderRadius": 12,
                  "justifyContent": "center",
                  "paddingHorizontal": 6,
                  "paddingVertical": 2,
                },
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "fontSize": 10,
                  },
                  undefined,
                ]
              }
            >
              测试tag
            </Text>
          </View>,
          null,
          null,
        ],
        "style": Object {
          "flexDirection": "row",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#fff",
                      "fontSize": 10,
                    },
                    undefined,
                  ]
                }
              >
                测试tag
              </Text>,
              undefined,
            ],
            "style": Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#FF301A",
                "borderRadius": 12,
                "justifyContent": "center",
                "paddingHorizontal": 6,
                "paddingVertical": 2,
              },
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "测试tag",
                "ellipsizeMode": "tail",
                "style": Array [
                  Object {
                    "color": "#fff",
                    "fontSize": 10,
                  },
                  undefined,
                ],
              },
              "ref": null,
              "rendered": "测试tag",
              "type": [Function],
            },
            undefined,
          ],
          "type": [Function],
        },
        null,
        null,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
